pkg/service/healthserver/healthserver.go

```diff
@@ -193,7 +193,7 @@ func (h *httpHealthHTTPServerFactory) newHTTPHealthServer(port uint16, svc *Serv
 				logfields.ServiceName:                svc.Service.Name,
 				logfields.ServiceNamespace:           svc.Service.Namespace,
 				logfields.ServiceHealthCheckNodePort: port,
-			}).Error("ListenAndServe failed for service health server")
+			}).Error("ListenAndServe failed for service health server, ensure that 'enable-health-check-nodeport' option is set to false if 'kube-proxy-replacement' is set to 'partial' mode")
```

Comment:
nit: Some errors returned by ListenAndServe might be legitimate failures.

I would prefer that we checked for `strings.Contains(err.Error(), "bind: address already in use")` and then have an error message along the lines of:
`This error may be caused by kube-proxy and cilum-agent both trying to serve a service health check. Please ensure that the 'enable-health-check-nodeport' option is set to false in cilium-agent if you are running kube-proxy intentionally.`

This is important for users where `kube-proxy-replacement=probe`. Users with `kube-proxy-replacement=partial` will already have the service health check disabled by default and will not see this message.

Reply:
Agreed, makes sense. I will try to address your comments.

Reply:
@gandro I have addressed your changes. Do you think we need to add an option through helm.
Or else can we directly document this cilium-agent option in the docs?